### PR TITLE
Don't throw on FIN from upgraded connections

### PR DESCRIPTION
**Pull request by halter73**

### CODE REVIEW

**Comment by davidfowl** [comment]
@dotnet-bot test OSX 10.12 Debug Build please

**Comment by davidfowl** [comment]
@dotnet-bot test OSX 10.12 Release Build please

**Comment by halter73** [comment]
@dotnet-bot test OSX 10.12 Release Build

**Comment by halter73** [comment]
@dotnet-bot test OSX 10.12 Debug Build

**Comment by halter73** [comment]
@davidfowl Should we try to push this through ask mode for 2.1 RTM?

**Comment by davidfowl** [comment]
Yep

File: test/Kestrel.FunctionalTests/UpgradeTests.cs [hunk 1]
```diff
@@ -297,5 +297,42 @@ public class UpgradeTests : LoggedTest
...
+                    await duplexStream.CopyToAsync(Stream.Null);
+                    appCompletedTcs.SetResult(null);
+                }
+                catch (Exception ex)
+                {
+                    appCompletedTcs.SetException(ex);
+                    throw;
+                }
+
+            }, new TestServiceContext(LoggerFactory)))
```
Comment (BrennanConroy) [inline test/Kestrel.FunctionalTests/UpgradeTests.cs]: Could you change the formatting to make it obvious that the `using(var server) {` starts here?

Comment (halter73) [reply test/Kestrel.FunctionalTests/UpgradeTests.cs]: I'm keeping this consistent with the other tests in this class and Kestrel in general. If you think there's better formatting, feel free to submit a PR, but it's will be a big diff.

**Pull request merged**